src/Services/SubmitService/SubmitService.cs

```diff
@@ -79,21 +87,54 @@ public async Task<string> ProcessSubmission(MappingEntity mappingEntity, string
                 reference = answers.CaseReference;
             }
 
-            return _submissionServiceConfiguration.FakeSubmission 
-                ? ProcessFakeSubmission(mappingEntity, form, sessionGuid, reference) 
+            var submissionReference = _submissionServiceConfiguration.FakeSubmission
+                ? ProcessFakeSubmission(mappingEntity, form, sessionGuid, reference)
                 : await ProcessGenuineSubmission(mappingEntity, form, sessionGuid, baseForm, reference);
+
+            if (baseForm.Pages is not null && mappingEntity.FormAnswers.Pages is not null)
+            {
+                var journeyPages = baseForm.GetReducedPages(mappingEntity.FormAnswers);
+
+                var isAutoConfirm = journeyPages.Any(_ => _.Elements.Any(_ => _.Type.Equals(EElementType.Booking) && _.Properties.AutoConfirm.Equals(true)));
+                if (isAutoConfirm)
+                    await ConfirmBooking(mappingEntity, form, sessionGuid, baseForm, reference);
+            }
+
+            return submissionReference;
+        }
+
+        private async Task ConfirmBooking(MappingEntity mappingEntity, string form, string sessionGuid, FormSchema baseForm, string reference)
+        {
+            var bookingProperties = baseForm.Pages
+                .Where(page => page.Elements is not null)
+                .SelectMany(page => page.Elements)
+                .First(element => element.Type.Equals(EElementType.Booking)).Properties;
+
+            var bookingId = Convert.ToString(mappingEntity.FormAnswers.AllAnswers
+                .ToDictionary(x => x.QuestionId, x => x.Response)
+                .Where(_ => _.Key.Equals($"{bookingProperties.QuestionId}-{BookingConstants.RESERVED_BOOKING_ID}"))
```

Comment:
If u get the element rather then the Properties, u can access ReservedIdQuestionId without the need to do the string interpolation